src/node_sea.cc

```diff
@@ -0,0 +1,120 @@
+#include "node_sea.h"
+
+#include "env-inl.h"
+#include "node_external_reference.h"
+#include "node_internals.h"
+#include "node_union_bytes.h"
+#include "simdutf.h"
+#include "v8.h"
+
+#define POSTJECT_SENTINEL_FUSE "NODE_JS_FUSE_fce680ab2cc467b6e072b8b5df1996b2"
+#include "postject-api.h"
+
+#include <memory>
+#include <string_view>
+#include <tuple>
+#include <vector>
+
+#if !defined(DISABLE_SINGLE_EXECUTABLE_APPLICATION)
+
+using v8::Context;
+using v8::FunctionCallbackInfo;
+using v8::Local;
+using v8::Object;
+using v8::Value;
+
+namespace {
+
+const std::string_view FindSingleExecutableCode() {
+  static const std::string_view sea_code = []() -> std::string_view {
+    size_t size;
+#ifdef __APPLE__
+    postject_options options;
+    postject_options_init(&options);
+    options.macho_segment_name = "NODE_JS";
+    const char* code = static_cast<const char*>(
+        postject_find_resource("NODE_JS_CODE", &size, &options));
+#else
+    const char* code = static_cast<const char*>(
+        postject_find_resource("NODE_JS_CODE", &size, nullptr));
+#endif
+    return {code, size};
+  }();
+  return sea_code;
+}
+
+void GetSingleExecutableCode(const FunctionCallbackInfo<Value>& args) {
+  node::Environment* env = node::Environment::GetCurrent(args);
+
+  const std::string_view sea_code = FindSingleExecutableCode();
+
+  if (sea_code.empty()) {
+    return;
+  }
+
+  size_t expected_u16_length =
+      simdutf::utf16_length_from_utf8(sea_code.data(), sea_code.size());
+  auto out = std::make_shared<std::vector<uint16_t>>(expected_u16_length);
+  size_t u16_length =
+      simdutf::convert_utf8_to_utf16(sea_code.data(),
+                                     sea_code.size(),
+                                     reinterpret_cast<char16_t*>(out->data()));
+  out->resize(u16_length);
+
+  args.GetReturnValue().Set(
+      node::UnionBytes(out).ToStringChecked(env->isolate()));
```

Comment:
Doesn't the `UnionBytes` here go out of scope as soon as this function returns, and with it the last reference to `out`, which is separate (converted copy) from the long-lived SEA code?

Reply:
@joyeecheung good catch, fixed! PTAL!